problem_builder/mentoring.py

```diff
@@ -776,3 +780,570 @@ def workbench_scenarios():
         Scenarios displayed by the workbench. Load them from external (private) repository
         """
         return loader.load_scenarios_from_path('templates/xml')
+
+
+@XBlock.needs("i18n")
+@XBlock.wants('settings')
+class MentoringWithExplicitStepsBlock(XBlock, StudioEditableXBlockMixin, StudioContainerXBlockMixin):
```

Comment:
@antoviaque Thanks for the feedback! I DRYed out the changeset based on your suggestions (removed most of the duplicate code and factored the remaining code that was shared between the two mentoring blocks out into a shared superclass).